www/include/views/graphs/graph-periods.php

```diff
@@ -52,52 +52,53 @@
 $tpl = new Smarty();
 $tpl = initSmartyTpl($path, $tpl);
 
-function getGetPostValue($str)
-{
-    $value = null;
-    if (isset($_GET[$str]) && $_GET[$str]) {
-        $value = $_GET[$str];
-    }
-    if (isset($_POST[$str]) && $_POST[$str]) {
-        $value = $_POST[$str];
-    }
-    return urldecode($value);
+if (isset($_GET['chartId'])) {
+    $chartId = filter_var($_GET['chartId'], FILTER_SANITIZE_STRING);
```

Comment:
to avoid an undefined $chartId line 59:
```suggestion
$chartId = filter_var($_GET['chartId'] ?? null, FILTER_SANITIZE_STRING);
```